target/linux/ramips/base-files/etc/board.d/02_network

```diff
@@ -425,6 +426,10 @@ ramips_setup_macs()
 		lan_mac=$(mtd_get_mac_ascii factory lanmac)
 		wan_mac=$(mtd_get_mac_ascii factory wanmac)
 		;;
+	dlink,dwr-116-a)
+		lan_mac=$(jboot_config_read -m -i /dev/mtd5 -o 0xE000)
+		wan_mac=$(macaddr_add "$lan_mac" 1)
```

Comment:
`wan_mac` and `lan_mac` should be swapped and I suggest to use the `find_mtd_index` function.

Example based on board `wsr-1166` :
```
		local index="$(find_mtd_index "config")"
		wan_mac="$(jboot_config_read -m -i "/dev/mtd${index}" -o 0xE000)"
		lan_mac=$(macaddr_add "$wan_mac" -1)
```